File: LeetCode/122买卖股票的最佳时机Ⅱ.cpp
/***************贪心做法****************/
int maxProfit(int* prices, int pricesSize){
    int ans = 0;
    for (int i = 1; i < pricesSize; i++) {
        ans += fmax(0, prices[i] - prices[i - 1]);
    }
    return ans;
}

/***************动态规划*************/
int maxProfit(int* prices, int pricesSize){
    int dp[pricesSize][2];
    dp[0][0] = 0, dp[0][1] = -prices[0];
    for (int i = 1; i < pricesSize; i++) {
        dp[i][0] = fmax(dp[i - 1][0], dp[i - 1][1] + prices[i]);
        dp[i][1] = fmax(dp[i - 1][1], dp[i - 1][0] - prices[i]);
    }
    return dp[pricesSize - 1][0];
}

File: LeetCode/122买卖股票的最佳时机Ⅱ.go
func maxProfit(prices []int) (ans int) {
	for i := 1; i < len(prices); i++ {
		ans += int(math.Max(float64(prices[i]-prices[i-1]), 0))
	}
	return
}

/*
	每天都可以买卖股票，如何获得最大收益，可以把股票营收按照天为单位
	即前一天买入，今天卖出，因为总的收益肯定是与当前每一天收入相关的
	每一天的收入相当于局部最优，那么局部最优加起来就是整体最优解
*/